examples/data_augmentation/reinforcement/main.py

```diff
@@ -0,0 +1,350 @@
+# limitations under the License.
+"""
+Example of building a reinforcement learning based,
+data augmentation enhanced sentence classifier
+based on pre-trained BERT model.
+"""
+import argparse
+import functools
+import logging
+import os
+
+import torch
+import torch.nn.functional as F
+import texar.torch as tx
+from transformers import BertForMaskedLM
+
+from examples.data_augmentation.reinforcement.config \
+    import config_data, config_classifier
+from examples.data_augmentation.reinforcement.utils import model_utils
+from forte.models.da_rl import MetaAugmentationWrapper, TexarBertMetaModule
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '--pretrained-model-name', type=str, default='bert-base-uncased',
+    choices=tx.modules.BERTEncoder.available_checkpoints(),
+    help="Name of the pre-trained downstream checkpoint to load.")
+parser.add_argument(
+    "--output-dir", default="output/",
+    help="The output directory where the model checkpoints will be written.")
+parser.add_argument(
+    "--do-train", action="store_true", help="Whether to run training.")
+parser.add_argument(
+    "--do-eval", action="store_true",
+    help="Whether to run eval on the dev set.")
+parser.add_argument(
+    "--do-test", action="store_true",
+    help="Whether to run test on the test set.")
+parser.add_argument(
+    '--augmentation-model-name', type=str, default='bert-base-uncased',
+    choices=tx.modules.BERTEncoder.available_checkpoints(),
+    help="Name of the pre-trained augmentation model checkpoint to load.")
+parser.add_argument(
+    '--num-aug', type=int, default=4,
+    help="number of augmentation samples when fine-tuning aug model")
+parser.add_argument(
+    '--classifier-pretrain-epoch', type=int, default=10,
+    help="number of epochs to pretrain the classifier")
+args = parser.parse_args()
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+logging.root.setLevel(logging.INFO)
+
+
+class RLAugmentClassifierTrainer:
+    def __init__(self):
+        self._prepare_data_iterator()
+        self._init_aug_model()
+        self._init_classifier()
+
+    def _prepare_data_iterator(self):
+        tx.utils.maybe_create_dir(args.output_dir)
+
+        # Loads data
+        num_train_data = config_data.num_train_data
+        self.num_train_steps = \
+            int(num_train_data / config_data.train_batch_size *
+                config_data.max_train_epoch)
+
+        train_dataset = tx.data.RecordData(
+            hparams=config_data.train_hparam, device=device)
+        val_dataset = tx.data.RecordData(
+            hparams=config_data.eval_hparam, device=device)
+        test_dataset = tx.data.RecordData(
+            hparams=config_data.test_hparam, device=device)
+        self.iterator = tx.data.DataIterator(
+            {"train": train_dataset,
+             "eval": val_dataset,
+             "test": test_dataset}
+        )
```

Comment:
Normally we use `train`, `dev`, `test`. `eval` is normally considered the same as `test`